src/alarms_agent.cpp

```diff
@@ -172,15 +172,15 @@ int main (int argc, char **argv)
     return 1;
   }
 
-  AlarmScheduler* alarm_scheduler = new AlarmScheduler(alarm_table_defs, snmp_notifications, hostname);
-  AlarmReqListener* alarm_req_listener = new AlarmReqListener(alarm_scheduler);
-
   init_alarmModelTable(*alarm_table_defs);
   init_ituAlarmTable(*alarm_table_defs);
   init_alarmActiveTable(local_ip);
  
   init_snmp_handler_threads("clearwater-alarms");
 
+  AlarmScheduler* alarm_scheduler = new AlarmScheduler(alarm_table_defs, snmp_notifications, hostname, SNMP::Agent::instance()->get_lock());
```

Comment:
Is this ordering change pertinent? If so, can you add a comment?

Reply:
Yes.  I've added a comment.